commit1

renvema opened the pull request:
-add product;
-add checking password;

Have a nice day )

Commit:
commit1
-add product;
-add checking password;

Commit:
commit2
-add product;
-add checking password;
-add product table;

Have a nice day )

Commit:
commit3 part2
-add Factory with Singlton;
-fix email, if passwords aren't equals;
-make form on general page;

Have a nice day )

Sofasmile commented on .idea/artifacts/onlineshoptomcat_war_exploded.xml:
```diff
@@ -0,0 +1,13 @@
+<component name="ArtifactManager">
```
what about .gitignore?

renvema replied:
Sorry, I don't know which files I must upload for checking, so I didn't add .gitignore

Sofasmile replied:
I will check all files where you write code or which you change

SevostyanovGerman replied:
gitignore

Sofasmile commented on pom.xml:
```diff
@@ -0,0 +1,34 @@
+    </build>
+
+
+</project>
```
add a blank line?

Sofasmile replied:
?

renvema replied:
ok

Sofasmile commented on src/main/java/controller/AllUserServlet.java:
```diff
@@ -0,0 +1,25 @@
+@WebServlet("/users")
+public class AllUserServlet extends HttpServlet {
+
+    private static final UserService userService =new UserServiceImpl();
```
```suggestion
    private static final UserService userService = new UserServiceImpl();
```
Ctrl+Alt+L

renvema replied:
done

Sofasmile commented on src/main/java/controller/ProductAddServlet.java:
```diff
@@ -0,0 +1,28 @@
+
+@WebServlet(value = "/add")
+public class ProductAddServlet extends HttpServlet {
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
https://mate-academy.github.io/style-guides/java/java.html#s4.4-column-limit

renvema replied:
done

Sofasmile commented on src/main/java/controller/ProductAddServlet.java:
```diff
@@ -0,0 +1,28 @@
+        Double price = Double.parseDouble(req.getParameter("price"));
+
+        Product user = new Product(1L, title, description,price);
+        System.out.println("Кладем товар в корзину");
```
better use System.out.println for testing and then delete it

renvema replied:
done

Sofasmile commented on src/main/java/controller/UserRegistrationServlet.java:
```diff
@@ -0,0 +1,38 @@
+@WebServlet(value = "/register")
+public class UserRegistrationServlet extends HttpServlet {
+
+    private static final UserService userService =new UserServiceImpl();
```
Ctrl+Alt+L

renvema replied:
done

Sofasmile commented on src/main/java/dao/impl/UserDaoImpl.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.List;
+
+public class UserDaoImpl implements UserDao {
+    private List<User> userList = new ArrayList<>();
```
better move this list in separate class which will be prototype of database with name "Storage"

renvema replied:
done

Sofasmile commented on src/main/java/dao/UserDao.java:
```diff
@@ -0,0 +1,12 @@
+import java.util.List;
+
+public interface UserDao {
+
```
cs

renvema replied:
done

Sofasmile commented on web/WEB-INF/web.xml:
```diff
@@ -0,0 +1,9 @@
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
+         version="4.0">
+
```
cs

Sofasmile replied:
remove redundant lines

renvema replied:
done

Commit:
commit4 HW19+HW20
-add Storage;
-change column limit;

Sofasmile commented on src/main/java/service/impl/ProductServiceImpl.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.List;
+
+public class ProductServiceImpl implements ProductService {
+    private static final ProductDao productDao = new ProductDaoImpl();
```
we need to use factory here
and use this class in controllers
the same about user

renvema replied:
done

Commit:
commit5 HW19+HW20+H21
-add button users in product;
-add button products in user;
-cs;
-move list in Storage;

Have a nice day ))

SevostyanovGerman commented on .idea/codeStyles/codeStyleConfig.xml:
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
-=-

renvema replied:
done

SevostyanovGerman commented on src/main/java/controller/AllProductServlet.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.List;
+
+@WebServlet("/tableproduct")
+public class AllProductServlet extends HttpServlet {
```
cs

renvema replied:
ok

SevostyanovGerman commented on src/main/java/controller/ProductAddServlet.java:
```diff
@@ -0,0 +1,40 @@
+public class ProductAddServlet extends HttpServlet {
+    private ProductDao productDao = ProductDaoFactory.getProductDao();
+    private Product product;
+    private Long id = 1L;
```
move to

renvema replied:
done

SevostyanovGerman commented on src/main/java/controller/ProductAddServlet.java:
```diff
@@ -0,0 +1,40 @@
+
+@WebServlet(value = "/add")
+public class ProductAddServlet extends HttpServlet {
+    private ProductDao productDao = ProductDaoFactory.getProductDao();
```
static final?

renvema replied:
done

SevostyanovGerman commented on src/main/java/controller/ProductAddServlet.java:
```diff
@@ -0,0 +1,40 @@
+
+        Product product = new Product(1L, title, description, price);
+        productService.addProduct(product);
+        id++;
```
-=-

SevostyanovGerman commented on src/main/java/controller/UserRegistrationServlet.java:
```diff
@@ -0,0 +1,45 @@
+
+@WebServlet(value = "/register")
+public class UserRegistrationServlet extends HttpServlet {
+    private UserDao userDao = UserDaoFactory.getUserDao();
```
static final

renvema replied:
done

SevostyanovGerman commented on src/main/java/controller/UserRegistrationServlet.java:
```diff
@@ -0,0 +1,45 @@
+public class UserRegistrationServlet extends HttpServlet {
+    private UserDao userDao = UserDaoFactory.getUserDao();
+    private User user;
+    private Long id = 1L;
```
-=-

renvema replied:
done

SevostyanovGerman commented on src/main/java/dao/ProductDao.java:
```diff
@@ -0,0 +1,11 @@
+
+import java.util.List;
+
+public interface ProductDao {
```
cs

renvema replied:
done

SevostyanovGerman commented on src/main/java/db/Storage.java:
```diff
@@ -0,0 +1,12 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
```
cs

renvema replied:
done

SevostyanovGerman commented on src/main/java/factory/ProductDaoFactory.java:
```diff
@@ -0,0 +1,15 @@
+import dao.ProductDao;
+import dao.impl.ProductDaoImpl;
+
+public class ProductDaoFactory {
```
cs

renvema replied:
done

Commit:
commit6 with gitignore

Commit:
commit7 with gitignore

Commit:
commit8 with gitignore

Commit:
commit with gitignore

Commit:
commit

Commit:
fixed untracked files

Commit:
commit with gitignore

Commit:
Merge remote-tracking branch 'origin/online' into online

Commit:
HW22 HW23

Commit:
fixed untracked files

Commit:
HW22 HW23 commit

Commit:
commit
-add filter;
-add ROLE USER,ADMIN

Commit:
commit
-add filter;
-add ROLE USER,ADMIN

SevostyanovGerman commented on src/main/java/controller/DeleteProductServlet.java:
```diff
@@ -0,0 +1,28 @@
+@WebServlet("/delete/product")
+public class DeleteProductServlet extends HttpServlet {
+
+    private ProductService productService = ProductServiceFactory.getInstance();
```
static final

SevostyanovGerman commented on src/main/java/controller/DeleteUserServlet.java:
```diff
@@ -0,0 +1,29 @@
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException, ServletException {
+        Long id = Long.valueOf(req.getParameter("id"));
+        User user=userService.getUsersById(id);
```
cs

SevostyanovGerman commented on src/main/java/controller/EditProductServlet.java:
```diff
@@ -0,0 +1,43 @@
+import java.io.IOException;
+
+@WebServlet(value = "/edit/product")
+public class EditProductServlet extends HttpServlet {
```
cs

SevostyanovGerman commented on src/main/java/controller/EditProductServlet.java:
```diff
@@ -0,0 +1,43 @@
+            product.setPrice(price);
+            req.setAttribute("allProduct", productService.getAll());
+            req.getRequestDispatcher("/products.jsp").forward(req, resp);
+        } catch (NumberFormatException | NullPointerException ex) {
```
logging?

SevostyanovGerman replied:
catch NPE?

SevostyanovGerman commented on src/main/java/controller/EditUserServlet.java:
```diff
@@ -0,0 +1,41 @@
+public class EditUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private User user;
```
why here?

SevostyanovGerman commented on src/main/java/dao/impl/ProductDaoImpl.java:
```diff
@@ -0,0 +1,40 @@
+                return product;
+            }
+        }
+        return null;
```
:(( Optional?

SevostyanovGerman commented on src/main/java/dao/impl/ProductDaoImpl.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    public void deleteProduct(Product product) {
```
may be by id?

SevostyanovGerman commented on src/main/java/dao/impl/UserDaoImpl.java:
```diff
@@ -0,0 +1,65 @@
+                return user;
+            }
+        }
+        return null;
```
Optional?

Commit:
commit H18_H24

Commit:
commit2 H18_H24

Sofasmile commented on pom.xml:
```diff
@@ -0,0 +1,66 @@
+            <scope>provided</scope>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/log4j/log4j -->
```
do we need this comment here?

Sofasmile commented on pom.xml:
```diff
@@ -0,0 +1,66 @@
+    </dependencies>
+
+
+    <build>
```
cs in all this file

Sofasmile commented on src/main/java/controller/EditProductServlet.java:
```diff
@@ -0,0 +1,64 @@
+            req.setAttribute("oldPrice", product.getPrice());
+            req.getRequestDispatcher("/edit_product.jsp").forward(req, resp);
+        } else {
+            logger.info("Productr is not found");
```
```suggestion
            logger.info("Product is not found");
```

Sofasmile commented on src/main/java/dao/impl/ProductDaoImpl.java:
```diff
@@ -0,0 +1,59 @@
+        return Storage.products;
+    }
+
+//    @Override
```
do we need this code here?

Sofasmile commented on src/main/java/dao/impl/UserDaoImpl.java:
```diff
@@ -0,0 +1,63 @@
+    @Override
+    public Optional<User> getUsersById(Long userId) {
+        List<User> allUsers = getAll();
+        return allUsers.stream()
```
better write:
```suggestion
        return Storage.users.stream()
```

Sofasmile commented on src/main/java/dao/impl/UserDaoImpl.java:
```diff
@@ -0,0 +1,63 @@
+
+        boolean result = false;
+
+        for (User user : Storage.users) {
```
maybe better use stream? read about anyMatch() method

Sofasmile commented on src/main/java/filter/AdminFilter.java:
```diff
@@ -0,0 +1,33 @@
+
+import model.User;
+
+import javax.servlet.*;
```
:)

Sofasmile commented on src/main/java/service/impl/ProductServiceImpl.java:
```diff
@@ -0,0 +1,31 @@
+
+public class ProductServiceImpl implements ProductService {
+
+    private static final ProductDao productDao = new ProductDaoImpl();
```
```suggestion
    private static final ProductDao productDao = ProductDaoFactory.getInstance();
```

Commit:
commit3 H18_H24

Commit:
commit4 H18_H24
-remove method userIsExist, because I used it in old filter;
-change autoimport;

Commit:
commit4 H25
-add radiobutton role

Commit:
commit5

Commit:
commit5
